src/benchmarklib/table_builder.hpp

```diff
@@ -0,0 +1,241 @@
+
+      column_definitions.emplace_back(name, data_type, is_nullable);
+    });
+    _table = std::make_shared<Table>(column_definitions, TableType::Data, chunk_size, use_mvcc);
+
+    // Reserve some space in the vectors
+    boost::hana::for_each(_value_vectors, [&](auto& values) { values.reserve(_estimated_rows_per_chunk); });
+    boost::hana::for_each(_null_value_vectors, [&](auto& null_values) {
+      if constexpr (std::decay_t<decltype(null_values)>::has_value) {
+        null_values.value().reserve(_estimated_rows_per_chunk);
+      }
+    });
+  }
+
+  std::shared_ptr<Table> finish_table() {
+    if (_current_chunk_row_count() > 0) {
+      _emit_chunk();
+    }
+
+    return _table;
+  }
+
+  // Types == DataTypes, we deduce again to get universal references, so we can call with lvalues
+  template <typename... Types>
+  void append_row(Types&&... new_values) {
+    auto values_tuple = boost::hana::make_tuple(std::forward<decltype(new_values)>(new_values)...);
+    BOOST_HANA_CONSTANT_ASSERT(boost::hana::size(_value_vectors) == boost::hana::size(values_tuple));
+
+    // Create tuples ([&values0, &null_values0, value0], [&values1, &null_values1, value1], ...)
+    auto value_vectors_and_null_value_vectors_and_values = boost::hana::zip_with(
+        [](auto& values, auto& null_values, auto&& value) {
+          return boost::hana::make_tuple(std::reference_wrapper(values), std::reference_wrapper(null_values),
+                                         std::forward<decltype(value)>(value));
+        },
+        _value_vectors, _null_value_vectors, values_tuple);
+
+    // The reason why we can't do the for_each stuff in zip_with is the following:
+    // boost::hana's higher order functions (like zip_with) do not guarantee the order of execution of a passed
+    // function f or even the number of executions of f. Therefore it should only be used with pure functions (no side
+    // effects). There are exceptions to that, for_each gives these guarantees and expects impure functions.
+
+    // Add the values to their respective value vector
+    boost::hana::for_each(value_vectors_and_null_value_vectors_and_values, [](auto& values_and_null_values_and_value) {
+      auto& values = values_and_null_values_and_value[boost::hana::llong_c<0>].get();
+      auto& null_values = values_and_null_values_and_value[boost::hana::llong_c<1>].get();
+      auto& maybe_optional_value = values_and_null_values_and_value[boost::hana::llong_c<2>];
+
+      constexpr bool column_is_nullable = std::decay_t<decltype(null_values)>::has_value;
+      auto value_is_null = table_builder::is_null(maybe_optional_value);
+
+      DebugAssert(column_is_nullable || !value_is_null, "cannot insert null value into not-null-column");
+
+      if (value_is_null) {
+        values.emplace_back();
+      } else {
+        // on failure: make sure template pack Types is the same that was passed to table_builder constructor
```

Comment:
Please revise comment, I don't see a failure nor something that "makes sure" anything here.